### ISSUE TITLE
blank map div of height 0

### ISSUE THREAD

**Issue by fcheng-cchnl** [Esri/jsapi-resources#492]
### IMPORTANT

- [X] My question is related to the samples and content within this repository. For all other issues, open a ticket with [Esri Technical Support](https://support.esri.com/) or post your question in the [community forum](https://community.esri.com/t5/arcgis-api-for-javascript/ct-p/arcgis-api-for-javascript).
- [X] I have [checked for existing issues](https://github.com/Esri/jsapi-resources/issues) to avoid duplicates. If someone has already opened an issue for what you are experiencing, please add a 👍 reaction and comment as necessary to the existing issue instead of creating a new one. You can also refer to this repo's [troubleshooting guide](../../TROUBLESHOOTING.md) for hints and suggestions.

### Actual behavior

This is a copy of the jsapi-angular-cli sample, modified for the Angular version we are using.
No errors are reported in the browser console. There are intentional console.log statements to trace if it's even reached the component initialization.

Element inspection shows the map div has 0 height but the css defines it as 100%.

Material CSS import is included but not used as we also use Material.

### Expected behavior

Map loads.

### Reproduction sample

https://github.com/fcheng-cchnl/angular15-esrimap.git

### Reproduction steps

1. Open Devtools.
2. Go to the Elements panel and expand app-esries
3. Expand the div with class="mydiv"
4. There will be text content "before mapViewNode" and "after mapViewNode"; the map div is in between. This div should show a height of 0.
5. Screenshot is attached
![zeroheightmapdiv](https://github.com/Esri/jsapi-resources/assets/36684947/ebaddbc8-2cb5-4334-94c9-10ca0c3f9fec)

### Reproduction browser

Chrome 118

### Operating System (check https://whatsmyos.com)

OS: Windows 10 Pro
Dell Precision 5820

**Comment by andygup** [comment]
Hi Fenton, have you tried setting the CSS on `AppComponent'? https://github.com/fcheng-cchnl/angular15-esrimap/blob/main/src/app/app.component.css

**Comment by fcheng-cchnl** [comment]
Hi Andy, I tried putting the import in all 3 locations one at a time (styles.css, app.component.css, esries.component.css) and still get the same result.

**Comment by andygup** [comment]
Okay, try using `vh` and `vw` instead of percentages. So, try something like this in `esries.component.css`:

```css
@import 'https://js.arcgis.com/4.27/@arcgis/core/assets/esri/themes/dark/main.css';
.esri-view {
  padding: 0;
  margin: 0;
  height: 100vh;
  width: 100vw;
}
#mapViewNode {
  padding: 0;
  margin: 0;
  height: 100vh;
  width: 100vw;
}
.mydiv {
  display: flex;
  height: 100vh;
  width: 100vw;
}
```

**Comment by fcheng-cchnl** [comment]
Hi Andy, I tried that with the sample and our full app and the map seems to load properly now. I'm looking at how to remove the double scroll bar now in the actual app.

**Comment by andygup** [comment]
Awesome, if using `vh` and `vw` solved your original issue as related to height 0, please go ahead and close this out.

TL;DR - setting div height/widget as percentages is sometimes tricky, as to when that size was calculated in the components CSS life-cycle. The rendering won't proactively validate if the height or width percentage of the parent element is zero. And, it doesn't take into account the [viewport](https://developer.mozilla.org/en-US/docs/Web/CSS/Viewport_concepts) size.

**Comment by fcheng-cchnl** [comment]
Thanks Andy!
